### PR TITLE
Absolute modding

### PR DESCRIPTION
**Pull request by felixlinker**
Modding stats for modules now happens via entering absolute values as opposed to percentage changes. This allows us to display those change rates as ingame.

Preview:
image

### CODE REVIEW

File: src/app/shipyard/Ship.js [hunk 1]
```diff
@@ -1187,38 +1182,35 @@ export default class Ship {
...
+    let slots = this.standard.concat(this.internal, this.hardpoints);
+    // TODO: create class for slot and also add slot.get
+    // handle unladen mass
+    unladenMass += chain(slots)
+      .map(slot => slot.m ? slot.m.get('mass') : null)
+      .filter()
+      .reduce((sum, mass) => sum + mass)
+      .value();
+
+    // handle fuel capacuty
```
Comment (willyb321) [inline src/app/shipyard/Ship.js]: slight nitpick, capacity*

File: src/app/components/Modification.jsx [hunk 2]
```diff
@@ -38,22 +39,8 @@ export default class Modification extends TranslatedComponent {
...
-      scaledValue = -9999;
-      value = -99.99;
-    }
-
-    let m = this.props.m;
-    let ship = this.props.ship;
-    ship.setModification(m, name, scaledValue, true);
-
+    let { m, name, ship } = this.props;
+    ship.setModification(m, name, value, true, true);
```
Comment (willyb321) [inline src/app/components/Modification.jsx]: at a glance, what purpose do the last 2 args serve?

File: src/app/shipyard/Module.js [hunk 3]
```diff
@@ -44,13 +44,7 @@ export default class Module {
...
     let result = this.mods  && this.mods[name] ? this.mods[name] : null;
 
-    // Calculate the percentage change for a synthetic value
-    if (STATS_FORMATING[name] && STATS_FORMATING[name].synthetic) {
-      const statGetter = this[STATS_FORMATING[name].synthetic];
-      let unmodifiedStat = statGetter.call(this, false);
-      let modifiedStat = statGetter.call(this, true);
-      result = (modifiedStat / unmodifiedStat - 1)  * 10000;
-    } else if ((!raw) && this.blueprint && this.blueprint.special) {
+    if ((!raw) && this.blueprint && this.blueprint.special) {
```
Comment (willyb321) [inline src/app/shipyard/Module.js]: why the double parenthesis around raw?

File: src/app/shipyard/Module.js [hunk 4]
```diff
@@ -135,14 +129,93 @@ export default class Module {
    * @param {Number} modified Whether to return the raw or modified value
    * @return {Number} The value queried
    */
-  _getValue(name, modified) {
+  get(name, modified = true) {
```
Comment (willyb321) [inline src/app/shipyard/Module.js]: not really a fan of setting args like that.

```
if (!modified) {
modified = true;
}

File: src/app/shipyard/Module.js [hunk 5]
```diff
@@ -260,13 +333,83 @@ export default class Module {
...
+  getUnitFor(name) {
+    const formatingOptions = STATS_FORMATING[name];
+    if (!formatingOptions || !formatingOptions.unit) {
+      if (formatingOptions.format && formatingOptions.format.startsWith('pct')) {
+        return 'pct';
+      }
+      return '';
+    }
+
+    return formatingOptions.unit;
```
Comment (willyb321) [inline src/app/shipyard/Module.js]: formatting with 2 `t`

File: src/app/shipyard/Module.js (hunk 3 above)
Comment (felixlinker) [reply src/app/shipyard/Module.js]: Actually I never touched that code.

File: src/app/shipyard/Module.js (hunk 4 above)
Comment (felixlinker) [reply src/app/shipyard/Module.js]: Uhm. Is a matter of preference but is less statements :)

File: src/app/shipyard/Module.js (hunk 3 above)
Comment (felixlinker) [reply src/app/shipyard/Module.js]: I wouldn't change this one as it's a wasted commit.

**Comment by felixlinker** [comment]
Also closes #249

**Pull request merged**